pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md

```diff
@@ -0,0 +1,114 @@
+# Generating my first single line diagrams - Guide 1
+
+In this guide we create a few single line diagrams of a test network with [powsybl-single-line-diagram](powsybl-single-line-diagram.md).
+
+
+First of all, we need to add a Maven dependency for the Network model and three additional ones for this example: two for the `Network` test case, one for simple logging capabilities:
+
+```xml
+<dependency>
+    <groupId>com.powsybl</groupId>
+    <artifactId>powsybl-iidm-impl</artifactId>
+    <version>3.7.1</version>
+</dependency>
+<dependency>
+    <groupId>com.powsybl</groupId>
+    <artifactId>powsybl-iidm-test</artifactId>
+    <version>3.7.1</version>
+</dependency>
+<dependency>
+    <groupId>com.powsybl</groupId>
+    <artifactId>powsybl-config-test</artifactId>
+    <version>3.7.1</version>
+</dependency>
+<dependency>
+    <groupId>org.slf4j</groupId>
+    <artifactId>slf4j-simple</artifactId>
+    <version>1.7.22</version>
+</dependency>
+```
+
+We can now load the node/breaker test `Network` we are interested in:
+```java
+Network network = FictitiousSwitchFactory.create();
+```
+
+In order to be able to generate the diagrams, we need to add the single line diagram core module dependency:
+```xml
+<dependency>
+    <groupId>com.powsybl</groupId>
+    <artifactId>powsybl-single-line-diagram-core</artifactId>
+    <version>1.7.2</version>
+</dependency>
+```
+
+Before generating diagrams we need to initialize a few layout parameters:
+
+```java
+// "Convergence" style component library
```

Comment:
"Convergence" is a RTE concept. We should rename this style or use a different one?

Reply:
You're right, I was thinking we should name it "Default" instead. I'll create an issue on powsybl-single-line-diagram

Reply:
We discussed it this morning with the other committers, they don't agree as the library is using the exact same components as RTE's Convergence and therefore keeping that name is decent. Nonetheless Convergence name will not appear in this guide soon:
- there should be soon a new default component library,
- in the next API changes I'm thinking of introducing a new method to generate a SVG without instantiating your own ComponentLibrary (the default ComponentLibrary will be chosen then).
Maybe we should continue this discussion in powsybl-single-line-diagram?

Reply:
[PR](https://github.com/powsybl/powsybl-single-line-diagram/pull/203) created on powsybl-single-line-diagram